addons/event_sale/models/event_registration.py

```diff
@@ -26,15 +27,18 @@ class EventRegistration(models.Model):
 
     @api.depends('is_paid', 'sale_order_id.currency_id', 'sale_order_line_id.price_total')
     def _compute_payment_status(self):
-        for record in self:
-            so = record.sale_order_id
-            so_line = record.sale_order_line_id
-            if not so or float_is_zero(so_line.price_total, precision_digits=so.currency_id.rounding):
-                record.payment_status = 'free'
-            elif record.is_paid:
-                record.payment_status = 'paid'
+        for so_line, registrations in self.grouped('sale_order_line_id').items():
+            if not so_line or float_is_zero(so_line.price_total, precision_digits=so_line.currency_id.rounding):
+                registrations.payment_status = 'free'
             else:
-                record.payment_status = 'to_pay'
+                paid_registrations = registrations.filtered('is_paid')
+                paid_registrations.payment_status = 'paid'
+                (registrations - paid_registrations).payment_status = 'to_pay'
+
+    @api.depends('payment_status')
+    def _compute_state(self):
+        for registration in self:
+            registration.state = 'open' if not registration.state and registration.payment_status != 'to_pay' else 'draft'
```

Comment:
Been stuck on this for a few minutes and this whole thing does not really feel right.

First, we have a stored computed field based on a computed non-stored field, which I don't believe is right.

Which means we should maybe store "payment_status", but then we have redundant information between "is_paid" and "payment_status". At that point we can probably remove is_paid an compute the payment_status based solely on the status of the SO and related account move (but is it even possible?).

Meaning we go back to part of your previous solution... so should we take the whole previous solution then? ( 😬 )

Another possibility would be to keep this, use a private business method that returns the payment_status and use it inside "_compute_state" and "_compute_payment_status", but it feels a bit dumb to compute the information twice.

Probably something to discuss at the farm tomorrow?